src/plugins/kernel_install/utils.sh

```diff
@@ -550,32 +550,85 @@ function do_uninstall()
   fi
 }
 
+# Element existence check
+# Checks if element belongs to given array
+#
+# @target_element element being checked
+# @array target array
+# returns 1 if element is in array and 0 otherwise
+function is_in_array()
+{
+  local target_element="$1"
+  local array=("${!2}")
+
+  for element in "${array[@]}"; do
+    if [[ "$element" == "$target_element" ]]; then
+      return 1
```

Comment:
By convention, 0 denotes true and not 0 denotes false. As the predicate is `is_in_array` invert the return values.